compiler_opt/es/blackbox_evaluator.py

```diff
@@ -97,12 +123,44 @@ def get_results(
       # update lists as work gets done
       _, not_done = concurrent.futures.wait(
           not_done, return_when=concurrent.futures.FIRST_COMPLETED)
-
     return futures
 
+  def get_results(
+      self, pool: FixedWorkerPool,
+      perturbations: list[bytes]) -> list[concurrent.futures.Future]:
+    # We should have _samples by now.
+    if not self._samples:
+      raise RuntimeError("Loaded samples are not available.")
+    return self._launch_compilation_workers(pool, perturbations)
+
   def set_baseline(self, pool: FixedWorkerPool) -> None:
-    del pool  # Unused.
-    pass
+    if self._baselines is not None:
+      raise RuntimeError('The baseline has already been set.')
+    self._load_samples()
+    results = self._launch_compilation_workers(pool)
+    self._baselines = super().get_rewards(results)
+
+  def get_rewards(
+      self, results: list[concurrent.futures.Future]) -> list[float | None]:
+    if self._baselines is None:
+      raise RuntimeError('The baseline has not been set.')
+
+    if len(results) != len(self._baselines):
+      raise RuntimeError(
+          'The number of results does not match the number of baselines.')
+
+    policy_results = super().get_rewards(results)
+
+    rewards = []
+    for i in range(len(policy_results)):
+      policy_result = policy_results[i]
+      baseline = self._baselines[i]
```

Comment:
`range(len())` is almost always a sign to use `enumerate()` or `zip()` instead.
```suggestion
    for policy_result, baseline in zip(policy_results, self._baselines, strict=True):
```